worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/BukkitImplAdapter.java

```diff
@@ -91,19 +94,20 @@ default void tickWatchdog() {
      *
      * @param location the location
      * @param state the block
-     * @param notifyAndLight notify and light if set
+     * @param sideEffectApplier side effects to apply
      * @return true if a block was likely changed
      */
-    boolean setBlock(Location location, BlockStateHolder<?> state, boolean notifyAndLight);
+    boolean setBlock(Location location, BlockStateHolder<?> state, SideEffectApplier sideEffectApplier);
 
     /**
      * Notifies the simulation that the block at the given location has
      * been changed and it must be re-lighted (and issue other events).
      *
      * @param position position of the block
      * @param previousType the type of the previous block that was there
+     * @param sideEffectApplier side effects to apply
      */
-    void notifyAndLightBlock(Location position, BlockState previousType);
+    void notifyAndLightBlock(Location position, BlockState previousType, SideEffectApplier sideEffectApplier);
```

Comment:
If we're breaking binary compat, might as well rename this to something better, probably the same as the `World` name.